sdk/core/core/src/az_span_builder.c

```diff
@@ -22,3 +22,56 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
   self->size += 1;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_span_builder_replace(
+    az_span_builder * const self,
+    size_t start,
+    size_t end,
+    az_span const span) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+  AZ_CONTRACT_ARG_NOT_NULL(&start);
+  AZ_CONTRACT_ARG_NOT_NULL(&end);
+
+  // Get current builder size
+  size_t const current_size = self->size;
+
+  // Start position must be lower than current builder size
+  AZ_CONTRACT(start <= current_size, AZ_ERROR_ARG);
+  // Start position must be less or equal than end position
+  AZ_CONTRACT(start <= end, AZ_ERROR_ARG);
+
+  // Get the size of the replaced span
+  size_t const replaced_size = end - start;
+
+  // current content must be greater or equal to what we want to replace
+  AZ_CONTRACT(current_size >= replaced_size, AZ_ERROR_ARG);
+
+  size_t const required_shift = current_size - end;
+  size_t const size_after_shift = current_size - replaced_size + span.size;
+
+  // if replace size is different than new content size, shift content
+  if (replaced_size != span.size) {
+    if (replaced_size < span.size) {
+      // validate theres enough space to shift right
+      AZ_CONTRACT(size_after_shift <= self->buffer.size, AZ_ERROR_ARG);
+
+      // shift right
```

Comment:
Can we not use the C runtime's memmove function? This function has been heavily optimized and takes of overlapping regions automatically.

Reply:
I would recommend to use `az_mut_span_move` which is a "safe" version of `memove`.